---
title: "Oman 2019 taxa table analysis"
output:
  html_document:
    css: stylesheet.css
    fig_caption: yes
    number_sections: no
    toc: yes
    toc_float: true
    toc_depth: 3
    code_folding: show
    df_print: paged
subtitle: "Source file: OM19_taxa_table_analysis.Rmd"
author: "Daniel Nothaft"
editor_options:
  chunk_output_type: inline
date: "`r Sys.Date()`"
---

# Setup

## Load packages
```{r setup, message=FALSE, warning=FALSE}
library(mctoolsr) # working with taxonomic data
library(tidyverse) # working with data frames
library(knitr) # printing things nicely

# global knitting options for automatic saving of all plots as .png and .pdf. Also sets cache directory.
knitr::opts_chunk$set(
  dev = c("png", "pdf"), fig.keep = "all",
  dev.args = list(pdf = list(encoding = "WinAnsi", useDingbats = FALSE)),
  fig.path = file.path("fig_output/", paste0(gsub("\\.[Rr]md", "/", knitr::current_input()))),
  cache.path = file.path("cache/", paste0(gsub("\\.[Rr]md", "/", knitr::current_input())))
)
```

## Load in map file and otutable
```{r load-data}
tax_table_fp <- 'data_output/20200521_tax_tab_DADA2_mctoolsr_OM19_Silva138_Bayes_full.txt'
map_fp <- 'data_raw/Summary_metadata/OM19_map16S_meta.txt'
input <- load_taxa_table(tax_table_fp, map_fp) #combine data to format for mctoolsr
```

# Overall trends and data curation

## look at reads per sample
Samples have more reads than the blank (at minimum 20X more reads), which is good.

Show read counts
```{r show-read-counts}
sort(colSums(input$data_loaded))
```

Plot read counts
```{r plot-read-counts}
# pull out metadata map
map <- input$map_loaded

# create data frame of reads
reads_per_sample_sums <- as.data.frame(colSums(input$data_loaded))

# create data frame of names of sample
reads_per_sample_names <- as.data.frame(colnames(input$data_loaded))

# combine reads with names to data frame of reads per sample
reads_per_sample <- cbind(reads_per_sample_sums, reads_per_sample_names)

# rename columns with more descriptive names
reads_per_sample <- rename(reads_per_sample, reads = "colSums(input$data_loaded)", sample_ID = "colnames(input$data_loaded)")

# add metadata
reads_per_sample <- left_join(reads_per_sample, map, by = "sample_ID") %>% select(sample_ID, reads, Description)

# create plot
read_tally <- reads_per_sample %>% ggplot(
  
  # plot data
  aes(x = reorder(sample_ID, -reads),
      y = reads,
      fill = Description)) +
  geom_bar(stat = "identity") +
  
  # axis styling
  scale_x_discrete(name = "sample ID") +
  
  # design
  theme_classic(base_size = 12) +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1))

# print plot
read_tally
```

## Filter out mitochondria, chloroplasts, eurkaryotes, and sequences not assigned taxonomy at the the domain level
```{r filter-extraneous-reads}
# Chloroplast not in dataset
input_filt <- filter_taxa_from_input(input, taxa_to_remove = c("Mitochondria", "Eukaryota"))
input_filt <- filter_taxa_from_input(input_filt, at_spec_level = 1, taxa_to_remove = "NA")
```

Show read counts of filtered data
```{r show-read-counts-filtered}
sort(colSums(input_filt$data_loaded))
```

## Focus on samples

```{r filter-for-samples}
# select desired samples
Oman_samples <- input_filt %>% filter_data(filter_cat = "sample_ID", filter_vals = c("Blank"))
```

Rarefy to lowest read count among samples
```{r rarefy}
Oman_samples = single_rarefy(Oman_samples, min(sort(colSums(Oman_samples$data_loaded))))

# extract metadata for desired samples
map_Oman_samples <- Oman_samples$map_loaded
```

# Taxonomic overview

Summarize taxonomy at species level
```{r summ-species}
# summarize taxonomic classification at species level
Oman_samples_sum <- summarize_taxonomy(Oman_samples, level = 7, report_higher_tax = TRUE)
```

Generate heat map
```{r heat-map, fig.height=8, fig.width=9}
Oman_samples_sum_higher_plot <- plot_ts_heatmap(Oman_samples_sum, map_Oman_samples, 0.005,  "well")

Oman_samples_sum_higher_plot +
  theme(axis.text.x = element_text(angle = 80, hjust = 1, vjust = 1, size = 7))
```

# CH$_4$-cycling taxa

## Sort data
### list taxa of interest
```{r input-CH4-cycle-taxa}
# input methane cycle
all_methane_taxa_oman <- c("Methanobacteria", "Methanomicrobia", "Methanopyrales", "Methanocellales", "Methanoplasmatales", "Methanosarcinales", "Methanomassiliicocc", "Methylococc", "Methylocystis", "Methylosinus", "Methylocella", "Methylocapsa", "Methylacidiphil", "Methylomirabilis", "ANME")
```

### filter for all CH$_4$-cycling taxa
```{r filter-for-CH4-cycle-taxa, warning=FALSE}
# filter for CH4-cycling taxa
Oman_samples_sum_all_methane_taxa <- filter_taxa_from_table(tax_table =  Oman_samples_sum, taxa_to_keep = all_methane_taxa_oman)

# get rownames of CH4 cycle taxa data frame
tax_long <- tibble(rownames(Oman_samples_sum_all_methane_taxa))
colnames(tax_long) <- "tax_long"

# Reshaping data so it is easier to work with

# make rownames into a column
Oman_samples_sum_all_methane_taxa <- bind_cols(tax_long, Oman_samples_sum_all_methane_taxa)

# print
Oman_samples_sum_all_methane_taxa
```

### show full taxonomy of CH$_4$-cycling taxa in easily searchable format
```{r show-CH4-taxonomy-searchable}
# gather data
Oman_samples_sum_all_methane_taxa_long_gathered <- Oman_samples_sum_all_methane_taxa %>%  gather(key = "sample_ID", value = "rel_abund", -tax_long)

# pretty print
Oman_samples_sum_all_methane_taxa_long_gathered %>% kable()
```

Add a shorthand name for taxa
```{r add-CH4-taxa-short}
Oman_samples_sum_all_methane_taxa <- Oman_samples_sum_all_methane_taxa  %>% mutate(tax_short = case_when(
  tax_long == "k__Archaea; p__Euryarchaeota; c__Methanobacteria; o__Methanobacteriales; f__Methanobacteriaceae; g__Methanobacterium; s__NA" ~ "genus Methanobacterium",
    tax_long == "k__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Methylococcales; f__Methylococcaceae; g__Methylococcus; s__NA" ~ "genus Methylococcus"
))

# put into easily searchable format
Oman_samples_sum_all_methane_taxa_gathered <- Oman_samples_sum_all_methane_taxa %>% gather(key = "sample_ID", value = "rel_abund", -tax_short, -tax_long)

# pretty print
Oman_samples_sum_all_methane_taxa_gathered %>% select(-tax_long) %>% kable()
```

Add metadata map to gathered CH$_4$-cycling taxa data frame
```{r add-metadata-map-CH4}
# join meta data map to CH4-cycling taxa data frame
gathered_w_map <- left_join(Oman_samples_sum_all_methane_taxa_gathered, map_Oman_samples, by = "sample_ID")
```

## Plot

### Prepare plot aesthetics

reorder factor levels so bar and legend plot in desired order
```{r reorder-factor-levels-CH4-taxa}
gathered_w_map$tax_short <- factor(gathered_w_map$tax_short, levels = c(
  "genus Methylococcus",
  "genus Methanobacterium"))
```

Set fill colors for taxa
```{r fill-colors-CH4-taxa}
# color-blind safe palette :
#F0E442 light yellow
#E69F00 mustard/gold
#D55E00 red
#CC79A7 magenta
#009E73 forest/mint green
#56B4E9 light blue
#0072B2 dark blue
#999999 grey
#000000 black

fill_taxa_named <- c(
  "genus Methylococcus" = "#E69F00",
  "genus Methanobacterium" = "#000000")
```

make formatting function to convert relative abundances to percent during plotting
```{r set-percent-formatter}
formatter100 <- function(x){ 
    x*100 
}
```

### Plot
```{r OM19-CH4-taxa-bar, fig.width = 5, fig.height = 4, dpi=300}
# create plot
OM19_CH4_taxa_bar <- gathered_w_map %>% ggplot(
  
  # plot data
  aes(x = well,
      y = rel_abund,
      fill = tax_short)) +
  geom_bar(stat = "identity") +
  
  # axis styling
  scale_y_continuous(name = "16S rRNA gene sequence percentage", labels = formatter100, expand = c(0.004, 0.002))+
  scale_x_discrete(name = NULL)+
  
  # symbol styling
  scale_fill_manual(name = "taxon", values = fill_taxa_named, labels = unname(latex2exp::TeX(c("genus \\textit{Methylococcus}",
  "genus \\textit{Methanobacterium}")))) +
  
  # design
  theme_classic(base_size = 12)+
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 9),
    legend.direction = "vertical",
    legend.text.align = 0
)

# print
OM19_CH4_taxa_bar
```

# Hydrogenophaga

## Sort data
### list taxa of interest
```{r input-Hydrogenophaga}
all_Hydrogenophaga_oman <- c("Hydrogenophaga")
```

### filter for all Hydrogenophaga
```{r filter-for-Hydrogenophaga, warning=FALSE}
# filter for Hydrogenophaga
Oman_samples_sum_all_Hydrogenophaga <- filter_taxa_from_table(tax_table =  Oman_samples_sum, taxa_to_keep = all_Hydrogenophaga_oman)

# get rownames of Hydrogenophaga data frame
tax_long <- tibble(rownames(Oman_samples_sum_all_Hydrogenophaga))
colnames(tax_long) <- "tax_long"

# Reshaping data so it is easier to work with

# make rownames into a column
Oman_samples_sum_all_Hydrogenophaga <- bind_cols(tax_long, Oman_samples_sum_all_Hydrogenophaga)

# print
Oman_samples_sum_all_Hydrogenophaga
```

### show full taxonomy of Hydrogenophaga in easily searchable format

```{r show-Hydrogenophaga-searchable}
# gather data
Oman_samples_sum_all_Hydrogenophaga_gathered <- Oman_samples_sum_all_Hydrogenophaga %>%  gather(key = "sample_ID", value = "rel_abund", -tax_long)

# pretty print
Oman_samples_sum_all_Hydrogenophaga_gathered %>% kable()
```

group taxa at genus level (Hydrogenophaga)

```{r}
# add grouped taxa name
Oman_samples_sum_all_Hydrogenophaga_grouped <- Oman_samples_sum_all_Hydrogenophaga %>% summarise_if(is.numeric, sum) %>% mutate(tax_group = "genus Hydrogenophaga")

Oman_samples_sum_all_Hydrogenophaga_grouped
```

### show Hydrogenophaga genus abundances in easily searchable format
```{r show-Hydrogenophaga-genus-searchable}
# gather genus-level Hydrogenophaga data frame
Oman_samples_sum_all_Hydrogenophaga_grouped_gathered <- Oman_samples_sum_all_Hydrogenophaga_grouped %>%  gather(key = "sample_ID", value = "rel_abund", -tax_group)

# pretty print
Oman_samples_sum_all_Hydrogenophaga_grouped_gathered %>% kable()
```

## Plot

### Prepare plot aesthetics
Add metadata
```{r add-metadata-Hydrogenophaga}
# add metadata
gathered_grouped_w_map_Hydrogenophaga <- left_join(Oman_samples_sum_all_Hydrogenophaga_grouped_gathered, map_Oman_samples, by = "sample_ID")

# print
gathered_grouped_w_map_Hydrogenophaga
```

### Plot
```{r OM19-Hydrogenophaga-bar, fig.width = 3, fig.height = 4, dpi=300}
# create plot
OM19_Hydrogenophaga_bar <- gathered_grouped_w_map_Hydrogenophaga %>% ggplot(
  
  # plot data
  aes(x = well,
      y = rel_abund)) +
  geom_bar(stat = "identity") +
  
  # axis styling
  scale_y_continuous(name = "16S rRNA gene sequence percentage", labels = formatter100)+
  scale_x_discrete(name = NULL)+
  
  # design
  theme_classic(base_size = 12)+
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 9),
    legend.direction = "vertical",
    legend.text.align = 0
)

# print
OM19_Hydrogenophaga_bar
```

# $\text{C}_{2+}$ Short-Chain Alkane (SCA)-consuming taxa
## Sort Data
### list taxa of interest
```{r list-SCA-taxa}
# input SCA-consuming taxa from Singh et al. (2017), Laso-Pérez et al. (2019), and Shennan et al. (2006)

all_SCA_taxa_oman <- c("BuS5", "Syntrophoarchaeum", "HotSeep-1", "Argoarchaeum", "ethanivorans", "Methanoliparia", "GoM", "Actinomyces", "Arthrobacter", "Brevibacterium", "Corynebacterium", "Gordonia", "Mycobacterium", "Nocardia", "Nocardiodes", "Rhodococcus", "Acinetobacter", "cepacia", "Pseudomonas", "Ralstonia")
```

Looking at a broader taxonomic level

The main point of this is to see if any close relatives to SCA-consuming taxa that were not directly identified in Silva 138. For example, Ca. Syntrophoarchaeum should be a clade within Methanomicrobia, and Ca. Methanoliparia should form its own class within Euryarchaeota (Laso-Pérez, 2019). As we will see, there are no unidentified Methanomicrobia or Euryarchaeota.

Regarding symbiotic SRB, I was interested if there was anything related to the Desulfosarcina/Desulfococcus (DSS) cluster within Deltaproteobacteria because this is the clade within which strain BuS5 (Laso-Pérez 2016) resides, but we see no evidence of closely related organisms in this Oman dataset.
```{r list-SCA-taxa-broader}
# input SCA-consuming
all_SCA_taxa_oman_broader <- c("Desulfobacteriaceae", "Desulfococcus", "Desulfosarcina", "Desulfobacterales", "Deltaproteobacteria", "Euryarchaeota")
```

### filter for all SCA-consuming taxa

Here, we see that there are no species-level identification of $\text{C}_{2+}$ short-chain alkane-oxidizing microbes. Many species in genus Pseudomonas do not perform short-chain alkane oxidation. Therefore, there is no conclusive detection of alkane-oxidizing microbes in this dataset. I still go on to plot these, for reference.

```{r filter-SCA-consuming-taxa, warning=FALSE}
# select SCA-consuming taxa
Oman_samples_sum_all_SCA_taxa <- filter_taxa_from_table(tax_table =  Oman_samples_sum, taxa_to_keep = all_SCA_taxa_oman )

# print SCA-consuming taxa
Oman_samples_sum_all_SCA_taxa
```

broader taxonomic level
```{r filter-SCA-consuming-taxa-broader, warning=FALSE}
# select SCA-consuming taxa
Oman_samples_sum_all_SCA_taxa_broader <- filter_taxa_from_table(tax_table =  Oman_samples_sum, taxa_to_keep = all_SCA_taxa_oman_broader)

# print SCA-consuming taxa
Oman_samples_sum_all_SCA_taxa_broader
```

Reshaping data so it is easier to work with
```{r reshape-SCA-data}
# get taxa names
tax_long_SCA <- tibble(rownames(Oman_samples_sum_all_SCA_taxa))
colnames(tax_long_SCA) <- "tax_long"

# make rownames into a column
Oman_samples_sum_all_SCA_taxa <- bind_cols(Oman_samples_sum_all_SCA_taxa, tax_long_SCA)

#print
Oman_samples_sum_all_SCA_taxa
```

### show full taxonomy of SCA cycling taxa in easily searchable format

```{r show-SCA-taxonomy-searchable}
# gather data
Oman_samples_sum_all_SCA_taxa_long_gathered <- Oman_samples_sum_all_SCA_taxa %>%  gather(key = "sample_ID", value = "rel_abund", -tax_long)

# pretty print
Oman_samples_sum_all_SCA_taxa_long_gathered %>% kable()
```

Add a shorthand name for SCA taxa
```{r add-SCA-taxa-short}
Oman_samples_sum_all_SCA_taxa <- Oman_samples_sum_all_SCA_taxa  %>% mutate(tax_short = case_when(
  tax_long == "k__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__NA" ~ "genus Pseudomonas"
))

# put into easily searchable format
Oman_samples_sum_all_SCA_taxa_gathered <- Oman_samples_sum_all_SCA_taxa %>% gather(key = "sample_ID", value = "rel_abund", -tax_short, -tax_long)

# pretty print
Oman_samples_sum_all_SCA_taxa_gathered %>% select(-tax_long) %>% kable()
```

### Show grouped taxonomy of SCA-cycling taxa in easily searchable format
```{r Oman_samples_sum_all_SCA_taxa_gathered}
gathered_w_map_SCA <- left_join(Oman_samples_sum_all_SCA_taxa_gathered, map_Oman_samples, by = "sample_ID")

gathered_w_map_SCA %>% select(well, tax_short, rel_abund) %>% kable()
```
## Plot
### Prepare plot aesthetics
Get some plot aesthetics ready

set color blind-safe palette
```{r set-cbPalette}
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
```

### Plot
```{r OM19_SCA_taxa_bar, fig.width = 5, fig.height = 4, dpi=300}
# create plot
OM19_SCA_taxa_bar <- gathered_w_map_SCA %>% ggplot(
  
  # plot data
  aes(x = well,
      y = rel_abund,
      fill = tax_short)) +
  geom_bar(stat = "identity") +
  
  # axis styling
  scale_y_continuous(name = "16S rRNA gene sequence percentage", labels = formatter100)+
  scale_x_discrete(name = NULL)+
  
  # symbol styling
  scale_fill_manual(name = "taxon", values = cbPalette, labels = unname(latex2exp::TeX(c(
  "genus \\textit{Pseudonomas}")))) +
  
  # design
  theme_classic(base_size = 12)+
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 9),
    legend.direction = "vertical",
    legend.text.align = 0
)

# print
OM19_SCA_taxa_bar
```